server/middlewares/passport.ts

```diff
@@ -35,18 +35,17 @@ export default function createMiddleware(providerName: string) {
             // get original host
             const stateString = ctx.cookies.get("state");
             const state = stateString ? parseState(stateString) : undefined;
-            const host = state?.host ?? ctx.hostname;
 
-            // form a URL object with the err.redirectUrl and replace the host
+            // form a URL object with the err.redirectPath and replace the host
             const reqProtocol =
               state?.client === Client.Desktop ? "outline" : ctx.protocol;
-            const requestHost = ctx.get("host");
+            const requestHost = state?.host ?? ctx.hostname;
             const url = new URL(
-              `${reqProtocol}://${requestHost}${redirectUrl}`
+              env.isCloudHosted
+                ? `${reqProtocol}://${requestHost}${redirectPath}`
+                : `${env.URL}${redirectPath}`
             );
 
-            url.host = host;
```

Comment:
This seems vestigial – why construct the URL above and then override the host immediately?